test/jdk/java/lang/System/LoggerFinder/SignedLoggerFinderTest/SignedLoggerFinderTest.java

```diff
@@ -0,0 +1,241 @@
+                "-Djdk.system.logger.level=DEBUG",
+                // useful for debug purposes
+                "-Djdk.logger.finder.error=DEBUG",
+                // enable logging to verify correct output
+                "-Djava.util.logging.config.file=" +
+                    Path.of(System.getProperty("test.src", "."), "logging.properties"),
+                "SignedLoggerFinderTest",
+                "no-init"));
+            if (mutliThreadLoad) {
+                cmds.add("multi");
+            }
+
+            try {
+                OutputAnalyzer outputAnalyzer = ProcessTools.executeCommand(cmds.stream()
+                                .filter(t -> !t.isEmpty())
+                                .toArray(String[]::new))
+                        .shouldHaveExitValue(0);
+                if (signJars) {
+                    outputAnalyzer
+                            .shouldContain("TEST LOGGER: [test_1, test]")
+                            .shouldContain("TEST LOGGER: [test_2, test]")
+                            .shouldContain(DNAME);
+                }
+
+            } catch (Throwable t) {
+                throw new RuntimeException("Unexpected fail.", t);
+            }
+        } else {
+            // set up complete. Run the code to trigger the recursion
+            mutliThreadLoad = args.length >=2 && args[1].equals("multi");
+            if (mutliThreadLoad) {
+                long sleep = new Random().nextLong(100L) + 1L;
+                System.out.println("multi thread load sleep value: " + sleep);
+                Runnable t = () -> {
+                    while(!testComplete) {
+                        // random logger call to exercise System.getLogger
+                        System.out.println("System.getLogger type:" +
+                            System.getLogger("random" + System.currentTimeMillis()).getClass().getName());
+                        try {
+                            Thread.sleep(sleep);
+                        } catch (InterruptedException e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                };
+                new Thread(t).start();
+            }
+            JarFile jf = new JarFile(jarPath1.toString(), true);
+            jf.getInputStream(jf.getJarEntry("loggerfinder/SimpleLoggerFinder.class"));
+            JarFile jf2 = new JarFile(jarPath2.toString(), true);
+            jf2.getInputStream(jf.getJarEntry("loggerfinder/SimpleLoggerFinder.class"));
+            Security.setProperty("test_1", "test");
+
+            // some extra sanity checks
+            assertEquals(System.LoggerFinder.getLoggerFinder().getClass().getName(),
+                    "loggerfinder.SimpleLoggerFinder");
+            Logger testLogger = Logger.getLogger("jdk.event.security");
+            assertEquals(testLogger.getClass().getName(), "java.util.logging.Logger");
+            testComplete = true;
```

Comment:
I believe these lines should be after `Security.setProperty("test_2", "test");`, to make sure the logger is loaded by the module that uses it. Also I was expecting some kind of checks to verify that the expected messages are indeed logged.